test/rpc_test.py

```diff
@@ -1062,6 +1062,29 @@ def test_call_method_on_rref(self):
 
         self.assertEqual(result, sum(vals))
 
+    @dist_init(setup_model_parallel=False)
+    @requires_process_group_agent("PROCESS_GROUP rpc backend specific test, skip")
+    def test_rpc_agent_destructor(self):
```

Comment:
Add a small comment here mentioning what we're testing:

"This test ensures that the rpc agent destructor safely shuts down the local RPC framework even if rpc.shutdown is not called"